internal/controllers/topology/cluster/conditions.go

```diff
@@ -103,6 +148,13 @@ func (r *Reconciler) reconcileTopologyReconciledCondition(s *scope.Scope, cluste
 				s.HookResponseTracker.AggregateMessage(),
 			),
 		)
+		v1beta2conditions.Set(cluster, metav1.Condition{
+			Type:   clusterv1.ClusterTopologyReconciledV1Beta2Condition,
+			Status: metav1.ConditionFalse,
+			Reason: clusterv1.ClusterTopologyReconciledHookBlockingV1Beta2Reason,
+			// TODO: Add a protection for messages continuously changing leading to Cluster object changes/reconcile.
+			Message: s.HookResponseTracker.AggregateMessage(),
```

Comment:
Wondering if we should have a sort of stale warning also here

Reply:
At the moment I only know which hooks are blocking the current reconcile.

I don't know which hook was blocking the previous reconciles. This has two dimensions: the kind of hook BeforeUpgrade, ... but also there could be multiple hooks e.g. someone could register 5 BeforeUpgrade hooks)

Reply:
Got it, I was thinking to look at the transition timestamp of the condition, but might be it is not accurate.
Let just keep this in the wish list for now

Reply:
Ah, I see. I wasn't thinking about that :)